tests/test_fields.py

```diff
@@ -0,0 +1,445 @@
+            self.assertEqual([], field_list.clean(empty_val))
+
+
+class FormFieldTests(SimpleTestCase):
+    class TestFormWithRequiredField(Form):
+        name = fields.CharField(required=True, max_length=100)
+
+    class TestFormWithoutRequiredField(Form):
+        name = fields.CharField(required=False, max_length=100)
+
+    def test_formfield_required(self):
+        form_field = FormField(form=self.TestFormWithRequiredField)
+
+        # TEST: valid input
+        valid_val = {'name': 'blah'}
+        self.assertEqual(valid_val, form_field.clean(valid_val))
+
+        # TEST: invalid input (values the FormField considers invalid)
+        invalid_vals = ['0', 1, datetime.datetime.now(), 'blah', {'blah'}, ['blah']]
+        expected_error = "({'name': [ValidationError(['Invalid value'])]}, None, None)"
+        for invalid_val in invalid_vals:
+            with self.assertRaisesMessage(RequestValidationError, expected_error):
+                log_input(invalid_val)
+                form_field.clean(invalid_val)
+
+        # TEST: required=True, form WITH required field - invalid input (empty value)
+        invalid_val = {'name': None}
+        expected_error = "({'name': [ValidationError(['This field is required.'])]}, None, None)"
+        with self.assertRaisesMessage(RequestValidationError, expected_error):
+            form_field.clean(invalid_val)
+
+        # TEST: required=True, form WITH required field - invalid input (unexpected dict key)
+        invalid_val = {'unexpected key': 'blah'}
+        expected_error = "({'name': [ValidationError(['This field is required.'])]}, None, None)"
+        with self.assertRaisesMessage(RequestValidationError, expected_error):
+            form_field.clean(invalid_val)
+
+        # TEST: required=True, form WITHOUT required field - unexpected dict key returns blanks with keys
+        # Django returns a normalized empty value when required=False rather than raising ValidationError
+        invalid_val = {'unexpected key': 'blah'}
+        form_field_no_required_fields = FormField(form=self.TestFormWithoutRequiredField)
+        self.assertEqual({'name': ''}, form_field_no_required_fields.clean(invalid_val))
+
+        # TEST: required=True - empty values throw an error
+        for empty_val in EMPTY_VALUES:
+            with self.assertRaisesMessage(ValidationError, "['This field is required.']"):
+                log_input(empty_val)
+                form_field.clean(empty_val)
+
+    def test_formfield_required_false(self):
+        form_field = FormField(form=self.TestFormWithRequiredField, required=False)
+
+        # TEST: valid input
+        valid_val = {'name': 'blah'}
+        self.assertEqual(valid_val, form_field.clean(valid_val))
+
+        # TEST: required=False - empty values return {}
+        """
+        for empty_val in EMPTY_VALUES:
+            self.assertEqual({}, form_field.clean(empty_val))
+        """
+        # AssertionError: {} != None
```

Comment:
Would it make more sense if this returned `{}`?: https://github.com/Sibyx/django_api_forms/blob/1bc2259f8e956ceab215f5be8701031830b91101/django_api_forms/fields.py#L84-L85

Reply:
Yeah, I guess that makes sense.

Reply:
I went ahead and changed this here: https://github.com/Sibyx/django_api_forms/pull/12/commits/ff26ba285ef165a99e0e0012c66e420b463f8767